tests/test_xarray.py

```diff
@@ -485,3 +485,32 @@ def test_mixed_grouping(chunk):
         fill_value=0,
     )
     assert (r.sel(v1=[3, 4, 5]) == 0).all().data
+
+
+@pytest.mark.parametrize("dtype_out", [np.float64, "float64", np.dtype("float64")])
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+@pytest.mark.parametrize("chunk", (True, False))
+def test_dtype(chunk, dtype, dtype_out, engine):
+    if chunk and not has_dask:
+        pytest.skip()
+
+    if chunk:
+        data = dask.array.ones((4, 12), dtype=dtype, chunks=(1, -1))
+    else:
+        data = np.ones((4, 12), dtype=dtype)
```

Comment:
How about using linspace instead? something like `np.linspace(0, 1, 48).reshape((4, 12))` ?
`mean` is not really challenged with constant values and I suppose the mean result can change a little depending on when the dtype changes.